core/src/main/java/com/graphhopper/speeds/WaySpeedsProvider.java

```diff
@@ -7,4 +7,6 @@
 public interface WaySpeedsProvider {
     Optional<SpeedKmByHour> speedForWay(long osmWayId);
     Optional<SpeedKmByHour> speedForRoadClass(RoadClass roadClass);
+    void setMetrics(Metrics metrics);
```

Comment:
Instead of a method, is not better `Metrics metrics` variable and let the implementation set on construction ?

Reply:
Yep, I thought about that...but the problem is that the logic deciding whether or not the speed is ignored is outside the interface WaySpeedProvider. Take a look [here](https://github.com/StuartApp/graphhopper-matrix/blob/develop/core/src/main/java/com/graphhopper/GraphHopperCustomSpeeds.java#L82)

Maybe I am wrong, but since the logic is outside the interface (WaySpeedsProvider), the interface needs to be used as a container of the Metrics, but the generation of the metrics itself is outside. Let me know you want to discuss it more in deep

Reply:
Finally, you are right...I will remove completely Metrics from interface WaySpeedsProvider, and it will be move to GraphhopperCustomSpeeds object